pytools/persistent_dict.py

```diff
@@ -698,129 +610,38 @@ def clear_in_mem_cache(self) -> None:
-
-                key_path = self._key_file(hexdigest_key)
-                value_path = self._contents_file(hexdigest_key)
-
-                self._write(value_path, value)
-                self._write(key_path, key)
-
-                logger.debug("%s: disk cache store [key=%s]",
-                        self.identifier, hexdigest_key)
-            except Exception:
-                cleanup_m.error_clean_up()
-                raise
-        finally:
-            cleanup_m.clean_up()
-
-    def fetch(self, key: K, _stacklevel: int = 0) -> Any:
-        hexdigest_key = self.key_builder(key)
-
-        (stored_key, stored_value) = self._fetch(hexdigest_key, 1 + _stacklevel)
-
-        self._collision_check(key, stored_key, 1 + _stacklevel)
-
-        return stored_value
-
-    def _fetch(self, hexdigest_key: str,  # pylint:disable=method-hidden
-               _stacklevel: int = 0) -> V:
-        # This is separate from fetch() to allow for LRU caching
-
-        # {{{ check path exists and is unlocked
-
-        item_dir = self._item_dir(hexdigest_key)
-
-        from os.path import isdir
-        if not isdir(item_dir):
-            logger.debug("%s: disk cache miss [key=%s]",
-                    self.identifier, hexdigest_key)
-            raise NoSuchEntryError(hexdigest_key)
-
-        lock_file = self._lock_file(hexdigest_key)
-        self._spin_until_removed(lock_file, 1 + _stacklevel)
-
-        # }}}
-
-        key_file = self._key_file(hexdigest_key)
-        contents_file = self._contents_file(hexdigest_key)
-
-        # Note: Unlike PersistentDict, this doesn't autodelete invalid entires,
-        # because that would lead to a race condition.
-
-        # {{{ load key file and do equality check
+    def store(self, key: K, value: V, _skip_if_present: bool = False) -> None:
+        keyhash = self.key_builder(key)
+        v = pickle.dumps((key, value))
 
         try:
-            read_key = self._read(key_file)
-        except Exception as e:
-            self._warn(f"{type(self).__name__}({self.identifier}) "
-                    f"encountered an invalid key file for key {hexdigest_key}. "
-                    f"Remove the directory '{item_dir}' if necessary. "
-                    f"(caught: {type(e).__name__}: {e})",
-                    stacklevel=1 + _stacklevel)
-            raise NoSuchEntryInvalidKeyError(hexdigest_key)
-
-        # }}}
-
-        logger.debug("%s: disk cache hit [key=%s]",
-                self.identifier, hexdigest_key)
-
-        # {{{ load contents
+            self.conn.execute("INSERT INTO dict VALUES (?, ?)", (keyhash, v))
+        except sqlite3.IntegrityError:
+            if not _skip_if_present:
+                raise ReadOnlyEntryError("WriteOncePersistentDict, "
+                                         "tried overwriting key")
+
+    def _fetch(self, keyhash: str) -> Tuple[K, V]:  # pylint:disable=method-hidden
+        # This method is separate from fetch() to allow for LRU caching
```

Comment:
Does the separation still make sense? There's no more LRU AFAICT. At least this comment can probably go.

Reply:
LRU should still be there, see line 605 in this file.